### PR TITLE
Initial Import

### PR DESCRIPTION
**Pull request by flyofsky**
Add default project for TizenDriver
This project support appium on Tizen devices.

### CODE REVIEW

File: lib/commands/find.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+let helpers = {}, extensions = {};
+
+/**
+ * Reason for isolating doFindElementOrEls from findElOrEls is for reusing findElOrEls
+ * across android-drivers (like appium-uiautomator2-driver) to avoid code duplication.
+ * Other android-drivers (like appium-uiautomator2-driver) need to override doFindElementOrEls
+ * to facilitate findElOrEls.
+ */
+helpers.doFindElementOrEls = async function (params) {
```
Comment (rookiejava) [inline lib/commands/find.js]: check comment again

File: gulpfile.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+"use strict";
+
+let gulp = require('gulp'),
+    boilerplate = require('appium-gulp-plugins').boilerplate.use(gulp);
+
+boilerplate({
+  build: 'appium-tizen-driver',
+  jscs: false,
+  e2eTest: {android: true},
```
Comment (rookiejava) [inline gulpfile.js]: ?

File: lib/tizen-bootstrap.js [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+      this.uiautomatorPkg = 'uiautomator-0.1.1-1.i686';
+      this.uiautomatorRpm = 'uiautomator-0.1.1-1.i686.rpm';
+    }
+
+    this.isRpmInstalled = await this.sdb.isRpmInstalled(this.uiautomatorPkg);
+    if (!this.isRpmInstalled) {
+      await this.installService(this.uiautomatorRpm, rpmPath);
+    }
+
+    //await this.sdb.killProcess('uiautomator');
```
Comment (rookiejava) [inline lib/tizen-bootstrap.js]: remove

File: lib/tizen-bootstrap.js [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+    if (this.sdb.curDeviceId.includes('emulator')) {
+      isEmulator = true;
+    }
+
+    if (isEmulator) {
+      this.uiautomatorPkg = 'uiautomator-0.1.1-1.i686';
+      this.uiautomatorRpm = 'uiautomator-0.1.1-1.i686.rpm';
+    }
+
+    this.isRpmInstalled = await this.sdb.isRpmInstalled(this.uiautomatorPkg);
```
Comment (rookiejava) [inline lib/tizen-bootstrap.js]: I don't think this approach (tightly coupled with certain package) is good. Do we need to insall uiautomator here?

File: lib/commands/general.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+commands.closeApp = async function () {
+  await this.sdb.forceStop(this.opts.appPackage);
+};
+
+commands.isStartedApp = async function (opts = {}) {
+  return await this.sdb.isStartedApp(this.opts.appPackage, opts);
+};
+
+
+commands.installService = async function (serviceRpm, serviceRpmPath, opts = {}) {
```
Comment (rookiejava) [inline lib/commands/general.js]: Need to reconsider providing install/uninstall (include setWriteFileSystem) here.

File: lib/commands/general.js [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    log.errorAndThrow(`Could not capture device date and time: ${err}`);
+  }
+};
+
+commands.inputText = async function (text) {
+  return await this.bootstrap.sendInputAction("inputText", [text]);
+};
+
+commands.pressHardWareKey = async function (keyCode) {
+  return await this.bootstrap.sendInputAction('pressHardWareKey', [keyCode]);
```
[the remark this answers is not in the document]
Comment (flyofsky) [reply lib/commands/general.js]: Done

File: lib/commands/general.js (hunk 5 above)
Comment (flyofsky) [reply lib/commands/general.js]: Done

File: README.md [hunk 7]
```diff
@@ -1,2 +1,19 @@
-# appium-tizen-driver
-Appium Tizen Driver is a test automation tool for Tizen devices. Appium Tizen Driver automates xamarin apps, tested on emulators and real devices. Appium Tizen Driver is part of the Appium mobile test automation tool.
+# Appium Tizen Driver
+
+Appium Tizen Driver is a test automation tool for Tizen devices. Appium Tizen Driver automates xamarin apps, tested on emulators and real devices. Appium Tizen Driver is part of the [Appium](https://github.com/appium/appium) mobile test automation tool.
```
Comment (rookiejava) [inline README.md]: xamarin apps -> .NET applications

Comment (flyofsky) [reply README.md]: Done

**Pull request merged**